### ISSUE TITLE
"custom_cflags" and "custom_optimization" Use Flags needed?

### ISSUE THREAD

**Issue by fxkrait** [InBetweenNames/gentooLTO#234]
Firefox and Thunderbird have the above Use-Flags, and I'm not sure if I should enable them or not.

For firefox I currently have `+lto -custom_cflags -custom_optimization`

Is this what I should have?
Thanks.

**Comment by ionenwks** [comment]
Firefox is generally very sensitive to optimizations, nearly anything you use can cause it to misbehave (for a while, it'd segfault even from `-march=native` on modern CPUs but that's been fixed). Doesn't help that the mozilla team doesn't even test it with recent GCC anymore and switched to clang for official builds (they use clang+LTO+PGO, gentoo doesn't let you enable PGO currently, albeit you just have to export `MOZ_PGO=1` and enjoy firefox taking 2x longer to build)

`custom-optimization` allows to use `-O3` instead of `-O2` which is actually bad with firefox+gcc (it won't segfaults which current version, but some features will misbehave and you'll get additional errors in the logs, with clang it defaults to `-O3` either way).

`custom-cflags` essentially does the same as what disabling flag-o-matic does, it just stops it from running `strip-flags()`, and running strip-flags is probably a good thing. Some things like graphite are probably harmless though, can't be too sure for the rest.

Don't know much for thunderbird (I don't use it), but I imagine it's in a similar situation.

tl;dr only use the `lto` USE flag (or if you want to make sure it runs as intended, use clang even)

**Comment by fxkrait** [comment]
Alright sounds good. I do recall when I tried some of these flags that the browser was misbehaving sometimes. I will leave them off. Thanks.
